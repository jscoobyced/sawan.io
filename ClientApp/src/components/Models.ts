import { match } from "react-router-dom";

export interface SelectOption {
    value: number;
    text: string;
}

export interface IdParam {
    id: string;
}
