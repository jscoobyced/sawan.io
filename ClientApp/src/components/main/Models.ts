import { NavigationMenuContent } from "../../services/Models";

export interface HeaderProps {
    navigationMenuContent: NavigationMenuContent;
    signIn: (isSignedIn: boolean) => void;
    isSignedIn: boolean;
}
