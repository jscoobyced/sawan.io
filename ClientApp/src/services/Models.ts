export enum Language {
    English = 0,
    French = 1
}

export interface Link {
    text: string;
    url: string;
    target?: string;
    title?: string;
}

export interface MainContent {
    navigationMenuContent: NavigationMenuContent;
    menuContent: MenuContent;
    footerContent: FooterContent;
}

export interface NavigationMenuContent {
    home: string;
    websiteName: string;
    applications: string;
    cryptoCurrency: string;
    healthMonitor: string;
    about: string;
    information: string;
    resume: string;
}

export interface BlogElement {
    id: number;
    blogDate: Date;
    article: string;
    articleTitle: string;
}

export interface BlogPage {
    articles: BlogElement[];
}

export interface MenuContent {
    links: Link[];
}

export interface FooterContent {
    copyright: string;
    credits: string;
    year: string;
}
