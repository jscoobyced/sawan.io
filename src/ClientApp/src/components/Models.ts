import { match } from "react-router-dom";

export interface SelectOption {
    value: number;
    text: string;
}
