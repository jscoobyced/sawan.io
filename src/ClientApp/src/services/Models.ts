import { Url } from "url";

export enum Language {
    English,
    French
}

export interface Link {
    text: string;
    url: string;
    target?: string;
    title?: string;
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
    menuContent: MenuContent;
    footerContent: FooterContent;
}

export interface NavigationMenuContent {
    home: string | string[];
    websiteName: string | string[];
    applications: string | string[];
    cryptoCurrency: string | string[];
    healthMonitor: string | string[];
    about: string | string[];
}

export interface HomeContent {
    articleTitle: string | string[];
    article: string | string[];
}

export interface MenuContent {
    links: Link[];
}

export interface FooterContent {
    copyright: string | string[];
    credits: string | string[];
    year: string | string[];
}