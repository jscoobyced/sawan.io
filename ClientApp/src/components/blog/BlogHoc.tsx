import React from 'react';
import { match } from 'react-router-dom';
import { ContentServiceFactory } from '../../services/ContentServiceFactory';
import { IContentService } from '../../services/IContentService';
import { BlogElement } from '../../services/Models';
import { AuthenticationFactory } from '../../utils/AuthenticationFactory';
import { DateUtil } from '../../utils/DateUtils';
import { IdParam } from '../Models';
import { Article } from './Article';
import { EditableArticle } from './EditableArticle';

export interface BlogHocProps {
    match: match<IdParam>;
}

export interface BlogState {
    blogElement: BlogElement;
}

export class BlogHoc extends React.Component<BlogHocProps, BlogState> {
    private readonly contentService: IContentService;

    public constructor(props: BlogHocProps, state: BlogState) {
        super(props, state);
        this.contentService = ContentServiceFactory.GetContentService();
        this.state = {
            blogElement: {
                article: '',
                articleTitle: '',
                blogDate: DateUtil.defaultDate(true),
                updateDate: DateUtil.defaultDate(true),
                id: 0
            }
        };
    }

    public componentDidUpdate(prevProps: BlogHocProps) {
        const articleId = +this.props.match.params.id;
        const prevArticleId = +prevProps.match.params.id;
        if (articleId !== prevArticleId) {
            this.doUpdate(articleId);
        }
    }

    public componentDidMount() {
        this.doUpdate(+this.props.match.params.id);
    }

    public updateBlog = (article: string, articleTitle: string, articleBlogDate: Date, id: number): Promise<boolean> => {
        const blogElement: BlogElement = {
            article,
            articleTitle,
            blogDate: articleBlogDate,
            updateDate: DateUtil.defaultDate(true),
            id
        };
        return this.contentService.saveBlogElement(blogElement);
    }

    public render() {
        const edit = this.props.match.params.edit === "edit";
        const isAdmin = AuthenticationFactory.getAuthentication() && AuthenticationFactory.getAuthentication().isAdmin();

        if (isAdmin && edit) {
            return (
                <EditableArticle
                    blogElement={this.state.blogElement}
                    update={this.updateBlog}
                />);
        } else {
            return (
                <Article
                    blogElement={this.state.blogElement}
                    backLink={true}
                    editLink={isAdmin}
                    ellipsis={false}
                />);
        }
    }

    private doUpdate(articleId: number) {
        if (articleId === 0) {
            return;
        }

        this.contentService.getBlogElement(articleId)
            .then(blogElement => {
                if (blogElement) {
                    this.setState({ blogElement });
                }
            });
    }
}
