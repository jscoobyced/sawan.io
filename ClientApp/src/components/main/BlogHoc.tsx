import React from 'react';
import { match } from 'react-router-dom';
import { ContentServiceFactory } from '../../services/ContentServiceFactory';
import { IContentService } from '../../services/IContentService';
import { BlogElement } from '../../services/Models';
import { IdParam } from '../Models';
import { Blog } from './Blog';

export interface BlogHocProps {
    match: match<IdParam>;
}

export interface BlogState {
    blogElement: BlogElement;
}

export class BlogHoc extends React.Component<BlogHocProps, BlogState> {
    private readonly contentService: IContentService;

    public constructor(props: BlogHocProps, state: BlogState) {
        super(props, state);
        this.contentService = ContentServiceFactory.GetContentService();
        this.state = {
            blogElement: {
                article: '',
                articleTitle: '',
                blogDate: new Date("2019-01-01T00:00:00+07:00"),
                id: 0
            }
        };
    }

    public componentDidMount() {
        this.contentService.getBlogElement(this.props.match.params.id)
            .then(blogElement => {
                if (blogElement) {
                    this.setState({ blogElement });
                }
            });
    }

    public render() {
        return <Blog blogElement={this.state.blogElement} />;
    }
}
