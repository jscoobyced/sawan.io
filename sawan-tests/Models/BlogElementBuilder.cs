namespace sawan.tests
{
    using System;

    public class BlogElementBuilder
    {
        public BlogElement Build()
        {
            return new BlogElement()
            {
                Article = "",
                ArticleTitle = "",
                BlogDate = DateTime.Now,
                UpdateDate = DateTime.Now,
                Id = 1
            };

        }
    }
}