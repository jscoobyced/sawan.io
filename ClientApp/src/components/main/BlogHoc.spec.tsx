import { shallow } from 'enzyme';
import React from 'react';
import { match } from 'react-router-dom';
import { IdParam } from '../Models';
import { BlogHoc } from './BlogHoc';

test('Blog is rendered empty when no blogElement', () => {
    const matchParam: match<IdParam> = {
        isExact: true,
        params: {
            id: '0'
        },
        path: '/',
        url: ''
    };
    const blog = shallow(<BlogHoc match={matchParam} />);
    expect(blog).toMatchSnapshot();
});