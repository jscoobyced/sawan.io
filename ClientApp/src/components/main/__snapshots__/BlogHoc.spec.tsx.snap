// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog is rendered empty when no blogElement 1`] = `
<Blog
  blogElement={
    Object {
      "article": "",
      "articleTitle": "",
      "blogDate": 2018-12-31T17:00:00.000Z,
      "id": 0,
    }
  }
/>
`;
